source/time_tables/departure_boards.cpp

```diff
@@ -270,11 +374,80 @@ void departure_board(PbCreator& pb_creator, const std::string& request,
         map_route_stop_point[route_point] = stop_times;
     }
 
-    render(pb_creator, response_status, map_route_stop_point, handler.date_time, handler.max_datetime,
-              calendar_id, depth);
+    render(pb_creator,
+           response_status,
+           map_route_stop_point,
+           map_route_first_point,
+           map_route_last_point,
+           handler.date_time,
+           handler.max_datetime,
+           calendar_id, depth);
 
     pb_creator.make_paginate(total_result, start_page, count, std::max(pb_creator.departure_boards_size(),
                                                                        pb_creator.stop_schedules_size()));
 }
+
+DateTime convert_duration_into_datetime(const datetime_type type,
```

Comment:
* The signature of this method is a bit weird.
Can't we return a struct as a response that contains both first and last, no ambiguity?

* Also, does it work for a Noctilien line? open=1am, close=5am, resquested=11pm > we want last_dt from that day, but first_dt to be the day after I guess (from other tests).
You probably want to use a complete requested_datetime, not just the date (as you want last_dt and first_dt to be always surrounding your requested_dt, first_dt and last_dt can be reverted)